test/python/test_rticonnextdds_data_access.py

```diff
@@ -125,9 +108,16 @@ def test_get_boolean_as_number(self, populated_input):
   def test_set_boolean_as_number(self, test_output, test_input):
     test_output.instance.set_number("my_optional_bool", 1)
     test_output.write()
-    self.wait_for_data(test_input)
+    wait_for_data(test_input)
     assert test_input[0]["my_optional_bool"] == True
 
+  def test_numeric_string(self, test_output, test_input):
+    test_output.instance["my_string"] = "1234"
+    sample = send_data(test_output, test_input)
+    # A side effect of CON-139: __getitem__ automatically parses strings and
+    # returns them as numbers if they represent a number
+    assert sample["my_string"] == 1234
```

Comment:
Is this behaviour a side effect or desireable? If not we could consider fixing it in the binding (check type of retValue -> if its a number call str() on it).